decorating.rb

```diff
@@ -1,27 +1,33 @@
-class Nameable 
-    def correct_name
-        return raise NotImplementedError
-    end
+# frozen_string_literal: true
+
+# Represents a nameable entity.
+class Nameable
+  def correct_name
+    raise NotImplementedError
+  end
 end
 
+# Represents a decorator for a Nameable object.
 class Decorator < Nameable
-    def initialize(nameable)
-      @nameable = nameable
-    end
-  
-    def correct_name
-      @nameable.correct_name
-    end
+  def initialize(age, parent_permission: true, name: "Unknown")
+    super(age, parent_permission: parent_permission, name: name)
+  end
```

Comment:
- [ ] Great job working on this milestone. In the constructor a nameable object from params should be assigned to an instance variable. I can see that you did the correct thing anc changed it. Kindly revert back.
![image](https://github.com/LDouglasOT/school-library/assets/62849872/fad75377-e39c-469b-84bb-cb753d3c623f)